scipy/interpolate/_bspl.pyx

```diff
@@ -412,3 +413,52 @@ def _norm_eq_lsq(const double[::1] x,
                 # ... and A.T @ y
                 for ci in range(rhs.shape[1]):
                     rhs[row, ci] = rhs[row, ci] + wrk[r] * y[j, ci] * wval
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def _make_design_matrix(const double[::1] x,
+                const double[::1] t,
+                int k,
+                kind):
+    '''
+    Returns a design matrix for BSpline object in "dense" or "CSR" formats.
+    
+    Parameters
+    ----------
+    x : 1-D array, shape(n,)
+        Values of x - coordinate of a given set of points (n >= 1).   
+    t : 1-D array, shape(nt,)
+        Vector of knots.
+    k : int
+        The maximum degree of spline.
+    kind : "CSR", optional
+        Default is ``"CSR"`` .
+        * ``"CSR"`` : Returns sparse matrix in CSR.
+
+    Returns
+    -------
+    Sparse matrix in CSR.
+    '''
+    n = len(x)
+    nt = len(t)
+    if kind == 'CSR':
+        data = np.zeros(n * (k + 1), dtype=float)
+        row_ind = np.zeros(n * (k + 1), dtype=int)
+        col_ind = np.zeros(n * (k + 1), dtype=int)
+        for i in range(n):
+            ind = find_interval(t, k, x[i], k - 1, 0)
+            vals = evaluate_all_bspl(t, k, x[i], ind)
```

Comment:
```suggestion
            ind = find_interval(t, k, x[i], k - 1, extrapolate=False)
            vals = evaluate_all_bspl(t, k, x[i], ind)
```
Could we replace `evaluate_all_bspl` by `_deBoor_D(...)` as the former is a Python function, the latter Cython?

Reply:
Yes, i think so.
Will replace in the next commit.